src/core/screen.c

```diff
@@ -1239,14 +1240,14 @@ get_window_surface (MetaWindow *window)
   /* Scale surface to fit current screen */
   if (width > height)
     {
-      max_size = MIN (MAX_PREVIEW_SIZE, current->rect.width / max_columns * MAX_PREVIEW_SCREEN_FRACTION);
+      max_size = MIN (MAX_PREVIEW_SIZE, MAX_PREVIEW_SCREEN_FRACTION * ((double) current->rect.width) / ((double) max_columns));
```

Comment:
Should it not be ?
```
- max_size = MIN (MAX_PREVIEW_SIZE, current->rect.width / max_columns * MAX_PREVIEW_SCREEN_FRACTION);
+ max_size = MIN (MAX_PREVIEW_SIZE, ((double) current->rect.width) /  MAX_PREVIEW_SCREEN_FRACTION * ((double) max_columns));
```

Reply:
Not sure if subtraction part is done before multiplication here in C.

Reply:
```
$ cat << EOF > polish_notation.c
#include <stdio.h>

int
main (void)
{
  printf ("%d\n", 8 / 4 * 2);
  printf ("%d\n", (8 / 4) * 2);
  return 0;
}
EOF
gcc -Wall polish_notation.c -o polish_notation
./polish_notation 
```
Result:
```
4
4
```

Reply:
The operators '*' and '/' have the same preference, so they operations are made from left to right.

Reply:
where is the subtraction?

Reply:
Opps, my translating mistake. I meant division.

> The operators '*' and '/' have the same preference, so the operations are made from left to right.

Thanks for explanation.